theano/gof/lazylinker_c.py

```diff
@@ -51,8 +51,17 @@ def try_reload():
             assert e.errno == errno.EEXIST
             assert os.path.isdir(location)
 
-    if not os.path.exists(os.path.join(location, '__init__.py')):
-        open(os.path.join(location, '__init__.py'), 'w').close()
+    init_file = os.path.join(location, '__init__.py')
+    if not os.path.exists(init_file):
+        try:
+            open(init_file, 'w').close()
+        except IOError as e:
+            if os.path.exists(init_file):
+                pass  # has already been created
+            else:
+                if os.path.exists(location):
+                    e.args += ('%s does not exist..' % location,)
```

Comment:
The error message do not match the condition.

Can you do something like:

e.args += ('%s exist?' os.path.exists(location),)

that would always give the information